docs/howto/object-storage/s3/presign.md

```diff
@@ -0,0 +1,167 @@
+     ETag      : 630f6e1bf441a0eee63f9cb06804dc79
+     Type      : file
+     Metadata  :
+       Content-Type       : application/pdf
+       X-Amz-Meta-Filename: bar.pdf
+       Content-Disposition: attachment
+     ```
+     You should thus set the `Content-Disposition` header with a different client.
+=== "s3cmd"
+    ```bash
+    s3cmd -c ~/.s3cfg-<region> modify \
+      --add-header 'Content-Disposition: attachment; filename="bar.pdf"'
+      s3://foo/bar.pdf
+    ```
+
+### Setting `Content-Disposition` on existing objects
+
+=== "aws"
+    To modify the `Content-Disposition` header of an existing object without downloading and re-uploading its contents, you must use `s3api copy-object`, with the object being its own copy source, and the metadata directive set to `replace`:
+    ```bash
+    aws --profile <region> \
+      --endpoint-url=https://s3-<region>.{{brand_domain}}:8080 \
+      s3api copy-object \
+      --copy-source <bucket-name>/<object-name>
+      --content-disposition 'attachment;filename="bar.pdf" \
+      --metadata-directive replace
+      --bucket <bucket-name> \
+      --key <object-name>
+    ```
+=== "mc"
+     The `mc` client does not support modifying object metadata in-place.
+=== "s3cmd"
+    `s3cmd` comes with a handy `modify --add-header` subcommand for updating object metadata in-place:
+    ```bash
+    s3cmd -c ~/.s3cfg-<region> modify \
+      --add-header 'Content-Disposition: attachment; filename="bar.pdf"'
+      s3://foo/bar.pdf
+    ```
+
+### Testing the `Content-Disposition` header
+
+Once you have set the `Content-Disposition` header on an object and created a pre-signed URL for it, you can test its functionality with the `curl -OJ` command:
+
+```console
+$ curl -f -OJ 'https://s3-kna1.{{brand_domain}}:8080/foo/bar.pdf?AWSAccessKeyId=07576783684248f7b2745e34356c6025&Expires=1673521496&Signature=%2Frm9nLV3moP%2FQz7aGCAnrESXjbk%3D'
+  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
+                                 Dload  Upload   Total   Spent    Left  Speed
+100 58503  100 58503    0     0   178k      0 --:--:-- --:--:-- --:--:--  178k
+curl: Saved to filename 'bar.pdf'
+```
+
+### Caution: do not abuse `Content-Disposition`
+
+Please keep in mind that there is nothing that keeps you from having two objects in the same bucket, with `Content-Disposition` on one object matching the object name of the other.
+
+For example, in the `foo` bucket, you may have
+
+* an object named `spam.pdf` with no `Content-Disposition` header, and
+* another object named `eggs.pdf`, with `Content-Disposition` set to include `filename="spam.pdf"`.
+
+If you were now to generate a pre-signed URL on `eggs.pdf` and downloaded it with `curl -OJ`, you would end up with a local file *named* `spam.pdf`, with the *contents* of `eggs.pdf`.
+This is obviously **not a good idea,** and you should avoid setting a `Content-Disposition` header that does not agree with the object name.
```

Comment:
Maybe the comma should not be in boldface? (Please!)

Reply:
So, the way I've been taught, if something is in bold or italics immediately preceding punctuation, then the punctuation is included in that emphasis. Is that wrong?

Reply:
Well, I had no idea there was a rule for that :/ Do not change the boldness, then!